src/XR/features/WebXRNearInteraction.ts

```diff
@@ -0,0 +1,592 @@
+            this._xrSessionManager.session.addEventListener("selectend", selectEndListener);
+        }
+    }
+
+    private _detachController(xrControllerUniqueId: string) {
+        const controllerData = this._controllers[xrControllerUniqueId];
+        if (!controllerData) {
+            return;
+        }
+        if (controllerData.squeezeComponent) {
+            if (controllerData.onSqueezeButtonChangedObserver) {
+                controllerData.squeezeComponent.onButtonStateChangedObservable.remove(controllerData.onSqueezeButtonChangedObserver);
+            }
+        }
+        if (controllerData.selectionComponent) {
+            if (controllerData.onButtonChangedObserver) {
+                controllerData.selectionComponent.onButtonStateChangedObservable.remove(controllerData.onButtonChangedObserver);
+            }
+        }
+        if (controllerData.onFrameObserver) {
+            this._xrSessionManager.onXRFrameObservable.remove(controllerData.onFrameObserver);
+        }
+        if (controllerData.eventListeners) {
+            Object.keys(controllerData.eventListeners).forEach((eventName: string) => {
+                const func = controllerData.eventListeners && controllerData.eventListeners[eventName as XREventType];
+                if (func) {
+                    this._xrSessionManager.session.removeEventListener(eventName as XREventType, func);
+                }
+            });
+        }
+        controllerData.pickIndexMeshTip?.dispose();
+        controllerData.hoverIndexMeshTip?.dispose();
+
+        // remove from the map
+        delete this._controllers[xrControllerUniqueId];
+        if (this._attachedController === xrControllerUniqueId) {
+            // check for other controllers
+            const keys = Object.keys(this._controllers);
+            if (keys.length) {
+                this._attachedController = keys[0];
+            } else {
+                this._attachedController = "";
+            }
+        }
+    }
+
+    private _generateNewHandTipMeshes() {
+        // populate information for near hover, pick and pinch
+        const meshCreationScene = this._options.useUtilityLayer ? this._options.customUtilityLayerScene || UtilityLayerRenderer.DefaultUtilityLayer.utilityLayerScene : this._scene;
+        var hoverIndexMeshTip = null;
+        var pickIndexMeshTip = null;
+
+        let createSphereMesh = (name: string, scale: number, visibility: boolean, sceneToUse: Scene): Nullable<AbstractMesh> => {
+            let resultMesh = null;
+            resultMesh = SphereBuilder.CreateSphere(name, { diameter: 1 }, sceneToUse);
+            resultMesh.scaling.set(scale, scale, scale);
+            resultMesh.isVisible = visibility;
+
+            return resultMesh;
+        };
+
+        hoverIndexMeshTip = createSphereMesh("IndexHoverSphere", this._hoverRadius, false, meshCreationScene);
+        pickIndexMeshTip = createSphereMesh("IndexPickSphere", this._pickRadius, false, meshCreationScene);

```

Comment:
nit: probably don't need to pass in visibility as a parameter since it's always false.